modules/serverless-logic-persistence-db-schema-initialization.adoc

```diff
@@ -0,0 +1,104 @@
+
+Flyway is managed by `kie-addons-quarkus-flyway` runtime module and it is disabled by default. To enable Flyway, you must configure it using one of the following methods:
+
+[id="serverless-logic-flyway-config-workflow-configmap_{context}"]
+== Flyway configuration in the workflow ConfigMap
+
+To enable Flyway in the workflow ConfigMap, you can add the following property:
+
+.Example of enabling Flyway in the workflow ConfigMap
+[source,yaml]
+----
+apiVersion: v1  
+kind: ConfigMap  
+metadata:  
+  labels:  
+    app: example-workflow  
+  name: example-workflow-props  
+data:  
+  application.properties: |  
+    kie.flyway.enabled = true
+----
+
+[id="serverless-logic-flyway-config-environment-variables_{context}"]
+== Flyway configuration using environment variables in the Workflow Container
+
+You can enable Flyway by adding an environment variable to the `spec.podTemplate.container` field in the `SonataFlow` CR by using the following example:
+
+.Example of enabling Flyway by using the workflow container environment variable
+[source,yaml]
+----
+apiVersion: sonataflow.org/v1alpha08  
+kind: SonataFlow  
+metadata:  
+  name: example-workflow  
+  annotations:  
+    sonataflow.org/description: Example Workflow  
+    sonataflow.org/version: 0.0.1  
+spec:  
+  podTemplate:  
+    container:  
+      env:  
+        - name: KIE_FLYWAY_ENABLED  
+          value: 'true'  
+  flow: ...
+----
+
+[id="serverless-logic-flyway-config-sonataflowplatform-properties_{context}"]
+== Flyway configuration using SonataFlowPlatform properties
+
+To apply a common Flyway configuration to all workflows within a namespace, you can add the property to the `spec.properties.flow` field of the `SonataFlowPlatform` CR shown in the following example:
+
+[NOTE]
+====
+This configuration is applied during workflow deployment. Ensure the Flyway property is set before deploying workflows.
+====
+
+.Example of enabling Flyway by using the `SonataFlowPlatform` properties
+[source,yaml]
+----
+apiVersion: sonataflow.org/v1alpha08  
+kind: SonataFlowPlatform  
+metadata:  
+  name: sonataflow-platform  
+spec:  
+  properties:  
+    flow:  
+      - name: kie.flyway.enabled  
+        value: true
+----
+
+[id="serverless-logic-manual-db-initialization-using-ddl_{context}"]
+== Initializing a manual database using DDL scripts
+
+If you prefer manual initialization, you must disable Flyway by ensuring the `kie.flyway.enabled` property is either not configured or explicitly set to `false`.
+
+* By default, each workflow uses a schema name equal to the workflow name. Ensure that you manually apply the schema initialization for each workflow.
+
+* If you are using the `SonataFlow` custom resource (CR) persistence configuration, you can specify a custom schema name.
+
+.Procedure
+
+. Download the DDL scripts from the {ServerlessLogic-DDL-Script-url} location.
```

Comment:
This URL gives the error: Path not found in group repository "maven.repository.redhat.com root repository" [id=root_repository].

Is this expected?

Reply:
Yes, this is expected. We have a GA next week. Once we are live it will be a working URL.